mapie/regression/regression.py

```diff
@@ -570,84 +452,34 @@ def fit(
             The model itself.
         """
         # Checks
-        self._check_parameters()
-        cv = check_cv(
-            self.cv, test_size=self.test_size, random_state=self.random_state
-        )
-        estimator = self._check_estimator(self.estimator)
-        agg_function = self._check_agg_function(self.agg_function)
-        X, y = indexable(X, y)
-        y = _check_y(y)
-        sample_weight = cast(Optional[NDArray], sample_weight)
-        self.n_features_in_ = check_n_features_in(X, cv, estimator)
-        sample_weight, X, y = check_null_weight(sample_weight, X, y)
-        self.conformity_score_function_ = check_conformity_score(
-            self.conformity_score
+        (estimator,
+         self.conformity_score_function_,
+         agg_function,
+         cv,
+         X,
+         y,
+         sample_weight) = self._check_fit_parameters(X, y, sample_weight)
+
+        self.estimator_ = EnsembleRegressor(
+            estimator,
+            self.method,
+            cv,
+            agg_function,
+            self.n_jobs,
+            self.random_state,
+            self.test_size,
+            self.verbose
         )
-        y = cast(NDArray, y)
-        n_samples = _num_samples(y)
-
-        # Initialization
-        self.estimators_: List[RegressorMixin] = []
-
-        # Work
-        if cv == "prefit":
-            self.single_estimator_ = estimator
-            y_pred = self.single_estimator_.predict(X)
-            self.k_ = np.full(
-                shape=(n_samples, 1), fill_value=np.nan, dtype=float
-            )
-        else:
-            cv = cast(BaseCrossValidator, cv)
-            self.k_ = np.full(
-                shape=(n_samples, cv.get_n_splits(X, y)),
-                fill_value=np.nan,
-                dtype=float,
-            )
-
-            self.single_estimator_ = fit_estimator(
-                clone(estimator), X, y, sample_weight
+        # Fit the prediction function
+        self.estimator_ = self.estimator_.fit(X, y, sample_weight)
+        y_pred = self.estimator_.predict_calib(X)
```

Comment:
SUGGESTION: your code works and it's excellent! I'm wondering here about the existence of `predict_calib`, which can cause problems (the user doesn't have access to this method separately from the `fit` method).  To prevent problem cases, I suggest you define a `fit_and_predict` method that returns `estimator, y_pred` in the `EnsembleEstimator` class. In doing so, I suggest redefining `predict_calib` as `_predict_calib`. This may be a minor change, but it's an important one in the logic of the `EnsembleEstimator` class.

Reply:
Just to clarify your suggestion you propose to : 

- call `predict_calib` in `EnsembleRegressor.fit(...)`
- change the signature of `EnsembleRegressor.fit(...)` to `EnsembleRegressor.fit_and_predict(...)`
- change the return type of `EnsembleRegressor.fit_and_predict(...)` to `Tuple[EnsembleRegressor, y_pred]`

Did I understand you correctly?

Reply:
- define a new method `EnsembleRegressor.fit_and_predict(...)` with `Tuple[EnsembleRegressor, y_pred]` as return type
- call `fit` and `predict_calib` in `EnsembleRegressor.fit_and_predict(...)`

Reply:
fit_and_predict is not used in sklearn, sometimes fit_transform is used but more for unsupervised learning